python/cudf/cudf/_lib/io/utils.pyx

```diff
@@ -40,6 +45,20 @@ cdef source_info make_source_info(list src) except*:
     # Otherwise src is expected to be a numeric fd, string path, or PathLike.
     # TODO (ptaylor): Might need to update this check if accepted input types
     #                 change when UCX and/or cuStreamz support is added.
+    elif isinstance(src[0], cudf.io.kafka.KafkaDatasource):
+        for key, value in src[0].kafka_configs.items():
+            kafka_confs[str.encode(key)] = str.encode(value)
+        consumer = new kafka_consumer(kafka_confs,
+                                      src[0].topic.encode(),
+                                      src[0].partition,
+                                      src[0].start_offset,
+                                      src[0].end_offset,
+                                      src[0].batch_timeout,
+                                      src[0].delimiter.encode())
```

Comment:
This is leaking currently.

Additionally, instead of generating the C object here, we should generate it when we construct the class instance how I mentioned.